### DISCUSSION TITLE
Error: Matrix has 312928 rows but found 45948 features. Try increasing `skip.feature`.

### DISCUSSION THREAD

**Discussion by carloslizama** [satijalab/seurat#4999] [General]
I got a mtx file from geodatabase. when Im using the readmtx, I got this error:

Error: Matrix has 312928 rows but found 45948 features. Try increasing `skip.feature`. 

Not sure what number I should use in the skip.feature.

Thanks

**Comment by saketkc** [comment]
The rows correspond to features (genes). 312928 looks like the number of cells in your file.  What is the output of `dim(readMM(/pathto/matrix.mtx)`? Most likely, the matrix being read is transposed version of what you want. You can also manually create the counts object using `readMM` and reading the gene names and cell name separately using read.table().

**Reply by carloslizama** [reply]
Sorry to take so long in replay but I was doing other stuff and I just start working again on it.
dim(readMM("C:/Users/14157/Desktop/2022/kaminski/matrix.mtx"))
I think the matrix looks:
[1]  45947 312928

dim(read_tsv("C:/Users/14157/Desktop/2022/kaminski/barcodes.tsv"))
Rows: 312927 Columns: 1   
I think the barcodes should be in a column format or not? 

dim(read_tsv("C:/Users/14157/Desktop/2022/kaminski/features.tsv"))
Rows: 45947 Columns: 2

**Comment by carloslizama** [comment]
I was able to fix the problem an build the matrix:
Following this instructions

library(Matrix)
matrix_dir = "/opt/sample345/outs/filtered_feature_bc_matrix/"
barcode.path <- paste0(matrix_dir, "barcodes.tsv.gz")
features.path <- paste0(matrix_dir, "features.tsv.gz")
matrix.path <- paste0(matrix_dir, "matrix.mtx.gz")
mat <- readMM(file = matrix.path)
feature.names = read.delim(features.path,
                           header = FALSE,
                           stringsAsFactors = FALSE)
barcode.names = read.delim(barcode.path,
                           header = FALSE,
                           stringsAsFactors = FALSE)
colnames(mat) = barcode.names$V1
rownames(mat) = feature.names$V1

However now I'm trying to build the seurat object, and I want to setup the orig.ident using the Barcodes ID however when Im use name.field =2, for some reason instead to setup the orig.ident with the barcodes ID just wrote seuratobject for all the cells.??

**Reply by saketkc** [reply]
You can just do `Idents(object) <- barcode.names$V1` if you want to set the barcode ID as the idents.

**Reply by carloslizama** [reply]
mm barcode.names&V1 will take the suffix barcode? or the full barcode name?

Basically this is my barcode: 001C_AAACCTGCATCGGGTC and I would like to set up my orig.ident using the 001C name.
My matrix is compilation of a lot of different samples but the name of the samples are in the barcodes. That the reason why I want to set up my orig.ident with this letters.

I need to subset the Seurat object based in the ID to extract the samples that I would like to analyzed.

Thanks

**Reply by saketkc** [reply]
You can extract `001C` using `stringr::str_split_fixed(barcode_IDs, pattern = "_", n=3)[,1]`

**Reply by carloslizama** [reply]
Do you know also if is possible to set-up the orig.ident something like that but instead to take the suffix take the prefix.

Can I do something like that when I'm creating the seurat object:

matobj <- CreateSeuratObject(mat, min.cells = 100, min.features = 200, assay = "RNA", project = stringr::str_split_fixed(barcode_IDs, pattern = "_", n=3)[,1]))

or I have to extract first from the barcode the ID in order to set-up later the orig.ident.

**Reply by saketkc** [reply]
`project` is attached to the entire object. You can always set your idents as `Idents(object) <- vector_of_identitities`. The suffix can be obtained using the same logic by replacing [,1] with [,3] or whatever order the suffix appears post splitting by _. See the [documentation](https://www.rdocumentation.org/packages/stringr/versions/0.6.2/topics/str_split_fixed).

**Reply by carloslizama** [reply]
Then I should setup:
ident(object) <-  stringr::str_split_fixed(barcode.names, pattern = "_", n=3)[,1]

Warning message:
In stri_split_regex(string, pattern, n = n, simplify = simplify,  :
  argument is not an atomic vector; coercing
